src/Service/Filesystem.php

```diff
@@ -492,4 +493,36 @@ protected function getTarExecutable()
         }
         throw new \RuntimeException("Tar command not found");
     }
+
+    /**
+     * @param $filename
+     * @return bool
+     */
+    public function fileExists($filename) {
+        return $this->fs->exists($filename);
+    }
+
+    /**
+     * @param string $filename
+     *
+     * @return array
+     */
+    public function readYamlFile($filename)
+    {
+        $contents = file_get_contents($filename);
+        if ($contents === false) {
+            throw new \RuntimeException('Failed to read file: ' . $filename);
+        }
+
+        return (array) Yaml::parse($contents);
+    }
+
+    /**
+     * @param $filename
+     * @param array $contents
+     */
+    public function createYamlFile($filename, array $contents) {
```

Comment:
If readYamlFile() is removed (see above comment) I don't think this method is necessary either (then Filesystem needn't depend on Symfony Yaml).